From 642218d8dd1ec79fa0c8db491fd46faa3ab026f7 Mon Sep 17 00:00:00 2001
From: Morgan Jones <me@numin.it>
Date: Tue, 21 Oct 2025 01:15:55 -0700
Subject: [PATCH 1/2] test-exec: use -P for allowed PKCS#11 library when
 starting agent

If we just loaded a PKCS#11 library, we should allow it so the
regression test can run.

Fixes: https://github.com/NixOS/nixpkgs/issues/453782
---
 regress/test-exec.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/regress/test-exec.sh b/regress/test-exec.sh
index 5b0c91f3faa..30b3da8709d 100644
--- a/regress/test-exec.sh
+++ b/regress/test-exec.sh
@@ -1023,6 +1023,9 @@ p11_ssh_add() {
 
 start_ssh_agent() {
 	EXTRA_AGENT_ARGS="$1"
+	if [ "$PKCS11_OK" = "yes" ]; then
+		EXTRA_AGENT_ARGS="${EXTRA_AGENT_ARGS} -P${TEST_SSH_PKCS11}"
+	fi
 	SSH_AUTH_SOCK="$OBJ/agent.sock"
 	export SSH_AUTH_SOCK
 	rm -f $SSH_AUTH_SOCK $OBJ/agent.log

From 5ae735db7d81b38ee059d63a4011291cb4456aef Mon Sep 17 00:00:00 2001
From: Morgan Jones <me@numin.it>
Date: Tue, 21 Oct 2025 01:50:23 -0700
Subject: [PATCH 2/2] test-exec: give more time for ssh-agent to start

---
 regress/test-exec.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/regress/test-exec.sh b/regress/test-exec.sh
index 30b3da8709d..56a7653b386 100644
--- a/regress/test-exec.sh
+++ b/regress/test-exec.sh
@@ -1034,7 +1034,7 @@ start_ssh_agent() {
 	    > $OBJ/agent.log 2>&1 &
 	AGENT_PID=$!
 	trap "kill $AGENT_PID" EXIT
-	for x in 0 1 2 3 4 ; do
+	for x in $(seq 15); do
 		# Give it a chance to start
 		${SSHADD} -l > /dev/null 2>&1
 		r=$?
